index.html

```diff
@@ -1787,6 +1951,310 @@ <h2>
+        A user agent can use these members to associate the web application
+        with [=file type=] on the operating system.
+      </p>
+      <p>
+        A <dfn>file type</dfn> can be defined by a [=mime type=] and/or [=file
+        extension=]. A file belongs to a file type if the OS determines it to
+        have a mime type and/or its name ends with a certain [=file
+        extension=]. A <dfn>file extension</dfn> is a string that start with a
+        "." and only contains [=valid suffix code points=]. A <dfn>valid suffix
+        code point</dfn> is a [=code point=] that is [=ASCII alphanumeric=],
+        U+002B (+), or U+002E (.). Additionally, [=file extensions=] are are
+        limited to a length of 16 code points.
+      </p>
+      <aside class="note">
+        <p>
+          Note: These code points were chosen to support most pre-existing file
+          formats. The vast majority of file extensions are purely
+          alphanumeric, but compound extensions (such as .tar.gz) and
+          extensions such as .c++ for C++ source code are also fairly common,
+          hence the inclusion of + and . as allowed code points.
+        </p>
+      </aside>
+      <aside class="note">
+        <p>
+          The selection of an app to handle a file launch if there are multiple
+          apps registerd for the file's [=file type=] is functionality left to
+          the operating system.
+        </p>
+      </aside>
+      <section>
+        <h3>
+          `action` member
+        </h3>
+        <p>
+          The [=file handler=]'s <code><dfn data-dfn-for=
+          "file handler">action</dfn></code> member is a <a>string</a> that
+          represents a relative URL of the [=manifest/start_url=] origin that
+          is [=manifest/within scope=] of a [=Document/processed manifest=].
+          This URL will be navigated to in the steps to [=execute a file
+          handler launch=].
+        </p>
+      </section>
+      <section>
+        <h3>
+          `name` member
+        </h3>
+        <p>
+          The [=file handler=]'s <code><dfn data-dfn-for=
+          "file handler">name</dfn></code> member is a <a>string</a> that
+          represents a name to be provided to the user's operating system for
+          use with "Open with..." UX flows. This MAY be replaced by the user
+          agent for privacy & security reasons.
+        </p>
+      </section>
+      <section data-cite="MIMETYPE">
+        <h3>
+          `accept` member
+        </h3>
+        <p>
+          The [=file handler=]'s <code><dfn data-dfn-for=
+          "file handler">accept</dfn></code> member is a <a>dictionary</a>
+          mapping [=MIME types=] to a list of [=file extensions=].
+        </p>
+        <p>
+          In order to [=register file handlers=], some operating systems
+          require [=MIME types=], some require [=file extensions=], and some
+          support both. Thus websites MUST always provide both for each [=file
```

Comment:
Right, so, if an app used:

`  'image/jpeg': ['.txt']`

then it would work with .txt files on Windows and not work with anything on Linux.

> can the file extension be made to win on Linux

No, because Linux (`xdg-mime`) only associates mime types to apps. If we tried to use some custom MIME type, e.g. telling `xdg-mime` that this app supports `application/x-appid-type1` the problem would be that we couldn't convince `xdg-mime` to map `foo.txt` to this custom MIME type; it would still map it to `text/plain`.